diff -up php-5.4.16/ext/interbase/config.m4.old php-5.4.16/ext/interbase/config.m4
--- php-5.4.16/ext/interbase/config.m4.old	2014-01-17 10:27:35.769189459 +0100
+++ php-5.4.16/ext/interbase/config.m4	2014-01-17 10:27:44.729127468 +0100
@@ -35,10 +35,11 @@ if test "$PHP_INTERBASE" != "no"; then
       -L$IBASE_LIBDIR
     ])
   ], [
-    -L$IBASE_LIBDIR
+    -L$IBASE_LIBDIR -lstdc++
   ])
   
   PHP_ADD_LIBRARY_WITH_PATH($IBASE_LIBNAME, $IBASE_LIBDIR, INTERBASE_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(stdc++, $IBASE_LIBDIR, INTERBASE_SHARED_LIBADD)
   PHP_ADD_INCLUDE($IBASE_INCDIR)
   AC_DEFINE(HAVE_IBASE,1,[ ])
   PHP_NEW_EXTENSION(interbase, interbase.c ibase_query.c ibase_service.c ibase_events.c ibase_blobs.c, $ext_shared)
diff -up php-5.4.16/ext/pdo_firebird/config.m4.old php-5.4.16/ext/pdo_firebird/config.m4
--- php-5.4.16/ext/pdo_firebird/config.m4.old	2014-01-17 10:28:04.912990400 +0100
+++ php-5.4.16/ext/pdo_firebird/config.m4	2014-01-17 10:28:09.928970914 +0100
@@ -42,12 +42,13 @@ if test "$PHP_PDO_FIREBIRD" != "no"; the
       $FIREBIRD_LIBDIR_FLAG
     ])
   ], [
-    $FIREBIRD_LIBDIR_FLAG
+    $FIREBIRD_LIBDIR_FLAG -lstdc++
   ])
  
   PHP_CHECK_PDO_INCLUDES
 
   PHP_ADD_LIBRARY_WITH_PATH($FIREBIRD_LIBNAME, $FIREBIRD_LIBDIR, PDO_FIREBIRD_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(stdc++, $FIREBIRD_LIBDIR, PDO_FIREBIRD_SHARED_LIBADD)
   PHP_ADD_INCLUDE($FIREBIRD_INCDIR)
   AC_DEFINE(HAVE_PDO_FIREBIRD,1,[ ])
   PHP_NEW_EXTENSION(pdo_firebird, pdo_firebird.c firebird_driver.c firebird_statement.c, $ext_shared,,-I$pdo_cv_inc_path)
