diff -Nrbu php-5.4.16/ext/pdo_dblib/dblib_stmt.c php-5.4.16-OK/ext/pdo_dblib/dblib_stmt.c
--- php-5.4.16/ext/pdo_dblib/dblib_stmt.c	2013-06-05 09:03:57.000000000 +0400
+++ php-5.4.16-OK/ext/pdo_dblib/dblib_stmt.c	2016-10-04 17:33:27.136889479 +0300
@@ -87,7 +87,6 @@
 
 	/* Cancel any pending results */
 	dbcancel(H->link);
-
 	efree(stmt->columns);
 	stmt->columns = NULL;
 	
@@ -98,8 +97,6 @@
 {
 	pdo_dblib_stmt *S = (pdo_dblib_stmt*)stmt->driver_data;
 
-	dblib_dblib_stmt_cursor_closer(stmt TSRMLS_CC);
-
 	efree(S);
 		
 	return 1;
@@ -131,6 +128,8 @@
 	
 	dbsetuserdata(H->link, (BYTE*) &S->err);
 	
+       dblib_dblib_stmt_cursor_closer(stmt TSRMLS_CC);
+       
 	if (FAIL == dbcmd(H->link, stmt->active_query_string)) {
 		return 0;
 	}
